src/CBF/CBF.jl

```diff
@@ -0,0 +1,605 @@
+            end
+
+            if startswith(line, "ACOORD")
+                for k in 1:parse(Int, strip(readline(io)))
+                    raw_coord = split(strip(readline(io)))
+                    @assert length(raw_coord) == 3
+                    (row_idx, var_idx) = (parse(Int, raw_coord[i]) + 1
+                        for i in 1:2) # CBF indices start at 0.
+                    coef = parse(Float64, raw_coord[end])
+                    push!(row_terms[row_idx], MOI.ScalarAffineTerm{Float64}(
+                        coef, scalar_vars[var_idx]))
+                end
+                continue
+            end
+
+            if startswith(line, "BCOORD")
+                for k in 1:parse(Int, strip(readline(io)))
+                    raw_coord = split(strip(readline(io)))
+                    @assert length(raw_coord) == 2
+                    row_idx = parse(Int, raw_coord[1]) + 1 # CBF indices start at 0.
+                    row_constants[row_idx] = parse(Float64, raw_coord[end])
+                end
+                continue
+            end
+
+            # PSD constraints.
+            if startswith(line, "PSDCON")
+                idx = 0
+                for k in 1:parse(Int, strip(readline(io)))
+                    side_dim = parse(Int, strip(readline(io)))
+                    push!(psd_side_dims, side_dim)
+                    push!(psd_row_starts, idx)
+                    idx += div(side_dim * (side_dim + 1), 2)
+                end
+                append!(psd_row_terms, Vector{MOI.ScalarAffineTerm{Float64}}()
+                    for i in 1:idx)
+                append!(psd_row_constants, zeros(idx))
+                continue
+            end
+
+            if startswith(line, "HCOORD")
+                for k in 1:parse(Int, strip(readline(io)))
+                    raw_coord = split(strip(readline(io)))
+                    @assert length(raw_coord) == 5
+                    (psd_idx, var_idx, i, j) = (parse(Int, raw_coord[i]) + 1
+                        for i in 1:4) # CBF indices start at 0.
+                    coef = parse(Float64, raw_coord[end])
+                    row_idx = psd_row_starts[psd_idx] + mat_to_vec_idx(i, j)
+                    push!(psd_row_terms[row_idx], MOI.ScalarAffineTerm{Float64}(
+                        coef, scalar_vars[var_idx]))
+                end
+                continue
+            end
+
+            if startswith(line, "DCOORD")
+                for k in 1:parse(Int, strip(readline(io)))
+                    raw_coord = split(strip(readline(io)))
+                    @assert length(raw_coord) == 4
+                    (psd_idx, i, j) = (parse(Int, raw_coord[i]) + 1
+                        for i in 1:3) # CBF indices start at 0.
+                    row_idx = psd_row_starts[psd_idx] + mat_to_vec_idx(i, j)
+                    psd_row_constants[row_idx] += parse(Float64, raw_coord[end])
+                end
+                continue
+            end
+        end
```

Comment:
If the code doesn't start with any of these lines, shouldn't we try to catch this as a corrupted file and throw an error?

Reply:
makes sense